app/main.py

```diff
@@ -139,13 +159,25 @@ def perform_topup(self) -> None:
                 else:
                     self.topups_added += 1
 
+
 def main() -> None:
-    lifeguard = Lifeguard()
-    while True:
-        lifeguard.poll()
+    try:
+        lifeguard = Lifeguard()
+        while lifeguard.poll():
+            print(f'Sleeping for {lifeguard.check_interval_minutes} minutes.')
+            time.sleep(lifeguard.check_interval_minutes * 60)
+        sys.exit(0)
+    except Exception as e:
+        while True:
```

Comment:
This could just be `traceback.print_exc(file=sys.stderr)`.

Shouldn't we be redirecting all logging to sys.stderr?

Reply:
If it's supposed to act like a command line app, then non-error logging should go to `stdout` & warning/error logging should go to `stderr`.  If it's supposed to act like a server app, then all logging should go to `stdout` period.